overlord/servicestate/quota_control.go

```diff
@@ -32,9 +35,40 @@ import (
 	"github.com/snapcore/snapd/snap/quota"
 	"github.com/snapcore/snapd/snapdenv"
 	"github.com/snapcore/snapd/strutil"
+	"github.com/snapcore/snapd/systemd"
+	"github.com/snapcore/snapd/timings"
 	"github.com/snapcore/snapd/wrappers"
 )
 
+var (
+	systemdVersion int
+)
+
+// TODO: move to a systemd.AtLeast() ?
+func checkSystemdVersion() error {
+	vers, err := systemd.Version()
+	if err != nil {
+		return err
+	}
+	systemdVersion = vers
+	return nil
+}
+
+func init() {
+	if err := checkSystemdVersion(); err != nil {
+		logger.Noticef("failed to check systemd version: %v", err)
+	}
+}
+
+func MockSystemdVersion(vers int) (restore func()) {
```

Comment:
(nitpick^2) Maybe a small doc string here like: `// MockSystemdVersion mocks the systemd version to the given version (e.g. 204). This is only available for unit test and will panic when run in production` ? But maybe overkill as it's pretty obvious I guess.

Reply:
added